rfcs/0145-doc-strings.md

```diff
@@ -0,0 +1,573 @@
+
+## Current State
+
+Currently, Nix does not have doc comments. Over time there evolved certain conventions, but they were never formally specified.
+Within this RFC, we want to clarify which convention we should use, harmonize and improve the format for doc-comments to achieve the envisioned goals.
+
+We use a tool called [nixdoc](https://github.com/nix-community/nixdoc) to use a 'doc-comment'- like functionality, which allows us to build a subset of static documentation of the `nixpkgs.lib` functions.
+
+The format of a doc-comment was, however, never specified. The nix community does not understand the rules enforced by `nixdoc`, which we believe is one of the reasons for its lack of adoption.
+
+Here is an example of the format understood by *nixdoc*:
+
+```nix
+# nixpkgs/lib/attrsets.nix
+
+{
+  /* <Description Field>
+     Example:
+     <Comprehensive Code example>
+     Type:
+     <Type signature>
+  */
+  AttrFunc =
+    # <Desribe arg1>
+    arg1:
+    # <Describe arg2>
+    arg2:
+    
+    # ... implementation
+}
+```
+
+## Current problems
+
+### Lack of having an agreed format
+
+The Nix community interprets the outer format and content of doc-comments differently. Within nixpkgs, several conventions and patterns need to be harmonized. It seems every directory decides upon their own convention; Or doesn't follow any. We immediately observe one clear thing: Most of comments utilize some form of markdown, even if they are not rendered.
+
+In general, the format for writing documentation strings is **not specified**.
+
+The *nixdoc*-tool requires a somewhat consistent format, but the basic format was never specified and cannot be enforced in parts of nixpkgs where nixdoc is currently not applied.
+
+Extending the scope of *nixdoc* is not the primary goal. Instead, we should find formal rules for writing *doc-comments*. Tools like *nixdoc* can then implement against this RFC instead of the format relying on nixdoc implementation details.
+
+### Impossible to differentiate from regular comments
+
+The format does not allow any distinction between doc comments and regular comments.
+
+Having a distinction would allow us to
+
+1. Find all comments that are part of the documentation
+2. Render them in the documentation format
+3. Connect the documentation to the exact places in the Nix code. 
+
+### References to the problems above
+
+#### nixpkgs - Dosctrings examples
+
+- [lib/attrsets](https://github.com/NixOS/nixpkgs/blob/master/lib/attrsets.nix)
+- [trivial-builders](https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/trivial-builders.nix)
+- [stdenv/mkDerivation](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix)
+- [nixos/lib/make-disk-image](https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-disk-image.nix)
+- more
+
+#### frameworks
+
+- [dream2nix/utils](https://github.com/nix-community/dream2nix/blob/main/src/utils/config.nix)
```

Comment:
What is this example trying to say? Lack of types signatures?

Reply:
This should showcase that nix-frameworks would also profit from a formalized doc-comment. 

In that example, they use single-line comments to write multiline documentation. 

That specific collection of links should show the inconsistency within the Nix ecosystem, which is a goal of this RFC.

Reply:
OK, so maybe mention this in the RFC itself? Otherwise it's not clear without reading these comments, IMO.